compiler-core/src/language_server/engine.rs

```diff
@@ -472,6 +490,52 @@ fn hover_for_pattern(pattern: &TypedPattern, line_numbers: LineNumbers) -> Hover
     }
 }
 
+fn hover_for_function_head(
+    fun: &Function<Arc<Type>, TypedExpr>,
+    line_numbers: LineNumbers,
+) -> Hover {
+    let empty_smolstr = SmolStr::from("");
+    let documentation = fun.documentation.as_ref().unwrap_or(&empty_smolstr);
```

Comment:
Does `.unwrap_or_default()` not work here? Or `.unwrap_or("".into())`

Reply:
`.unwrap_or_default()` was the first thing I'd tried since you were using it for all the results of `.get_documentation()`. That didn't work because `.documentation` itself is a `SmolStr`, and `&SmolStr` doesn't have `Default` implemented.

Trying some more stuff, the only other way I can see is

```rust
    let documentation = fun
        .documentation
        .as_ref()
        .map(|s| s.as_str())
        .unwrap_or_default();
```

which feels "cleaner" in some way, but also less readable?